nex-rs/src/server.rs

```diff
@@ -315,13 +315,15 @@ pub trait Server: EventHandler {
 
         if packet_type == PacketType::Disconnect {
             let addr = client.get_address();
-            self.kick(&mut clients, addr);
+            self.kick(addr).await;
         }
 
         Ok(())
     }
 
-    fn kick(&self, clients: &mut Vec<ClientConnection>, addr: SocketAddr) {
+    async fn kick(&self, addr: SocketAddr) {
```

Comment:
This will cause a deadlock since handle_socket_message has already acquired the lock and is a calling function of Server::kick

Reply:
Selected the wrong line by accident but it's the change to getting the mutex